src/transforms/detect_exceptions/mod.rs

```diff
@@ -0,0 +1,382 @@
+
+    /// Maximum number of lines to flush (0 means no limit). Default: 1000.
+    #[serde(default = "default_max_lines_num")]
+    pub max_lines: usize,
+}
+
+impl Default for DetectExceptionsConfig {
+    fn default() -> Self {
+        Self {
+            languages: vec![ProgrammingLanguages::All],
+            expire_after_ms: default_expire_after_ms(),
+            flush_period_ms: default_flush_period_ms(),
+            multiline_flush_interval_ms: default_multiline_flush_interval_ms(),
+            max_bytes: default_max_bytes_size(),
+            max_lines: default_max_lines_num(),
+            group_by: vec![],
+        }
+    }
+}
+
+fn default_programming_languages() -> Vec<ProgrammingLanguages> {
+    Vec::from([ProgrammingLanguages::All])
+}
+
+const fn default_expire_after_ms() -> Duration {
+    Duration::from_millis(30000)
+}
+
+const fn default_flush_period_ms() -> Duration {
+    Duration::from_millis(1000)
+}
+
+const fn default_multiline_flush_interval_ms() -> Duration {
+    Duration::from_millis(1000)
+}
+
+const fn default_max_bytes_size() -> usize {
+    0
+}
+
+const fn default_max_lines_num() -> usize {
+    1000
+}
+
+impl_generate_config_from_default!(DetectExceptionsConfig);
+inventory::submit! {
+    TransformDescription::new::<DetectExceptionsConfig>("detect_exceptions")
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "detect_exceptions")]
+impl TransformConfig for DetectExceptionsConfig {
+    async fn build(&self, _context: &TransformContext) -> crate::Result<Transform> {
+        DetectExceptions::new(self).map(Transform::event_task)
+    }
+
+    fn input(&self) -> Input {
+        Input::log()
+    }
+
+    fn outputs(&self, _: &schema::Definition) -> Vec<Output> {
+        vec![Output::default(DataType::Log)]
+    }
+
+    fn transform_type(&self) -> &'static str {
+        "detect_exceptions"
+    }
+}
+
+pub struct DetectExceptions {
+    accumulators: HashMap<Discriminant, TraceAccumulator>,
+    languages: Vec<ProgrammingLanguages>,
+    expire_after: Duration,
+    flush_period: Duration,
+    multiline_flush_interval: Duration,
+    max_bytes: usize,
+    max_lines: usize,
+    group_by: Vec<String>,
+}
+
+impl DetectExceptions {
+    pub fn new(config: &DetectExceptionsConfig) -> crate::Result<Self> {
+        if config.languages.is_empty() {
+            return Err("languages cannot be empty".into());
+        }
+        Ok(DetectExceptions {
+            accumulators: HashMap::new(),
+            languages: config.languages.clone(),
+            group_by: config.group_by.clone().into_iter().collect(),
+            expire_after: config.expire_after_ms,
+            multiline_flush_interval: config.multiline_flush_interval_ms,
+            max_bytes: config.max_bytes,
+            max_lines: config.max_lines,
+            flush_period: config.flush_period_ms,
+        })
+    }
+
+    fn consume_one(&mut self, output: &mut Vec<Event>, e: Event) {
+        let log_event = e.into_log();
```

Comment:
e.into_log() panics if e is not a LogEvent, is that what we want? There is try_into_log() that doesn't. It sure is not normal for a non-LogEvent to get routed here, but may be it's better to stay up and complain?

Reply:
This method in `impl TransformConfig for DetectExceptionsConfig {` ensures that this transform only gets `LogEvent`s
```
    fn input(&self) -> Input {
        Input::log()
    }
```
i also had same concern, and clarified it in discord